app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,43 @@
 class PostsController < ApplicationController
-  def index; end
+  before_action :find_post, only: [:show]
+  before_action :initialize_like
 
-  def show; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id])
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
```

Comment:
- [ ] Dear @pushkar1995, You have done great for the most part, but when I try to create a post, it simply does nothing. Investigating the console I notice it is logging an error about a redirection issue. Please address this issue and submit it for review. 

Note: This also prevents me from checking the functionality for the `like` and `comment` features, which are expected to happen on a post.  Please make sure this is in place before the next review request.

https://github.com/pushkar1995/blog-app/assets/69077061/32feaaa4-d3e2-45e2-9a19-f54d930ae44c

Reply:
@TadesseDev I have fixed the issue as mentioned by the reviewer.

Reply:
- Well reolved
<img width="978" alt="Screenshot 2023-11-11 at 12 11 05" src="https://github.com/pushkar1995/blog-app/assets/58692590/e73a6593-8726-478c-a583-bc13292bd9bc">